Sources/Widgets/Widgets3D/LineWidget/index.js

```diff
@@ -0,0 +1,177 @@
+import { distance2BetweenPoints } from 'vtk.js/Sources/Common/Core/Math';
+import stateGenerator from 'vtk.js/Sources/Widgets/Widgets3D/LineWidget/state';
+import macro from 'vtk.js/Sources/macro';
+import vtkAbstractWidgetFactory from 'vtk.js/Sources/Widgets/Core/AbstractWidgetFactory';
+import vtkArrowHandleRepresentation from 'vtk.js/Sources/Widgets/Representations/ArrowHandleRepresentation';
+import vtkPlanePointManipulator from 'vtk.js/Sources/Widgets/Manipulators/PlaneManipulator';
+import vtkSphereHandleRepresentation from 'vtk.js/Sources/Widgets/Representations/SphereHandleRepresentation';
+import vtkCubeHandleRepresentation from 'vtk.js/Sources/Widgets/Representations/CubeHandleRepresentation';
+import vtkConeHandleRepresentation from 'vtk.js/Sources/Widgets/Representations/ConeHandleRepresentation';
+import widgetBehavior from 'vtk.js/Sources/Widgets/Widgets3D/LineWidget/behavior';
+import vtkSVGCustomLandmarkRepresentation from 'vtk.js/Sources/Widgets/SVG/SVGCustomLandmarkRepresentation';
+import vtkPolyLineRepresentation from 'vtk.js/Sources/Widgets/Representations/PolyLineRepresentation';
+import { ViewTypes } from 'vtk.js/Sources/Widgets/Core/WidgetManager/Constants';
+// ----------------------------------------------------------------------------
+// Factory
+// ----------------------------------------------------------------------------
+
+const handleRepresentationType = {
+  SPHERE: 'sphere',
+  CUBE: 'cube',
+  CONE: 'cone',
+  ARROW: 'arrow',
+};
+
+function vtkLineWidget(publicAPI, model) {
+  model.classHierarchy.push('vtkLineWidget');
+
+  // --- Widget Requirement ---------------------------------------------------
+
+  // custom handles set in default values
+
+  const handleRepresentation = [0, 0, 0];
+  function detectHandleShape() {
```

Comment:
You should be able to simplified with something like this:
```
const handleRepresentation = [0, 0, 0];
  const shapeToRepresentation = {}
  shapeToRepresentation[handleRepresentationType.SPHERE] = vtkSphereHandleRepresentation;
  shapeToRepresentation[handleRepresentationType.CUBE] = vtkCubeHandleRepresentation;
  shapeToRepresentation[handleRepresentationType.CONE] = vtkConeHandleRepresentation;
  shapeToRepresentation[handleRepresentationType.ARROW] = vtkArrowHandleRepresentation;

  function detectHandleShape() {
    handleRepresentation[0] = shapeToRepresentation[model.shapeHandle1];
    if (!handleRepresentation[0]) {
      handleRepresentation[0] = vtkSphereHandleRepresentation;
    }
    handleRepresentation[1] = shapeToRepresentation[model.shapeHandle2];
    if (!handleRepresentation[1]) {
      handleRepresentation[1] = vtkSphereHandleRepresentation;
    }
    handleRepresentation[2] = vtkSVGCustomLandmarkRepresentation;
  }
```
I think that would be better to not use an array to store the representation. Maybe a map or something else.